desdeo_tools/scalarization/MOEADSF.py

```diff
@@ -0,0 +1,147 @@
+import abc
+from abc import abstractmethod
+from os import path
+from typing import List, Union
+
+import numpy as np
+
+
+class MOEADSFError(Exception):
+    """Raised when an error related to the MOEADSF classes is encountered.
+
+    """
+
+
+class MOEADSFBase(abc.ABC):
+    """A base class for representing scalarizing functions for the MOEA/D algorithm.
+    Instances of the implementations of this class should work as
+    function.
+
+    """
+
+    @abstractmethod
+    def __call__(self, objective_vector: np.ndarray, weight_vector: np.ndarray, reference_point: np.ndarray) -> Union[float, np.ndarray]:
```

Comment:
We should have a discussion regarding the nomenclature you are using. Specially "reference_point" which is used with a different meaning in different files. Once we have a common nomenclature, we can accept the pull request.